Make use of raw and RawInfo more consistent (#274)
quicwg/qlog: draft-ietf-quic-qlog-quic-events.md

    ; to support passing multiple at once
    ? count: uint16

    ; The RawInfo fields do not include the UDP headers, only the UDP payload
    ? raw: [+ RawInfo]

    ? datagram_ids: [+ uint32]
}
~~~
{: #transport-datagramssent-def title="TransportDatagramsSent definition"}

The "datagram_id" is a qlog-specific concept to allow tracking of QUIC packet

> coalescing inside UDP datagrams. Implementations can assign a per-endpoint

Reviewer: I realise this text was already there but why is it per-endpoint?

Author: Because there is no (obvious?) in-band way to share datagram-ids between individually logging endpoints? To keep them consistent, you'd have to have a stable hash or something?

unique ID to each datagram, and reflect this in other events to track QUIC
packets through processing steps.

Since QUIC implementations rarely control UDP logic directly, the raw data

> excludes UDP-level headers in all fields.

Reviewer: Consider moving this above the previous paragraph to match the ordering of the fields

    ; to support passing multiple at once
    ? count: uint16

    ; The RawInfo fields do not include the UDP headers, only the UDP payload
    ? raw: [+ RawInfo]

    ? datagram_ids: [+ uint32]
}
~~~
{: #transport-datagramsreceived-def title="TransportDatagramsReceived definition"}

For more details on "datagram_ids", see {{transport-datagramssent}}.

## datagram_dropped {#transport-datagramdropped}
Importance: Extra

When a UDP-level datagram is dropped. This is typically done if it does not
contain a valid QUIC packet. If it does, but the QUIC packet is dropped for

> other reasons, {{transport-packetdropped}} should be used instead.

Reviewer: This would just render as a section number right? That's not very reader-friendly.
```suggestion
other reasons, packet_dropped ({{transport-packetdropped}}) should be used instead.
```

    ? from: "user" / "application" / "transport" / "network" / text
    ? to: "user" / "application" / "transport" / "network" / text

    ? raw: RawInfo
}
~~~~
{: #transport-datamoved-def title="TransportDataMoved definition"}

This event does not utilize a "direction" field (with values "up" and "down")
to specify the data flow. This is because in some optimized implementations, data
might skip some individual layers. Additionally, using explicit "from" and "to"
fields is more flexible and allows the definition of other conceptual "layers"
(for example to indicate data from QUIC CRYPTO frames being passed to a TLS

> library or from HTTP/3 to QPACK).

Reviewer: Is this paragraph necessary? There are no events with a `direction` field. The purpose of from/too fields is already desdcribed in the earlier paragraphs. This text seems superflous to me.

### ConnectionCloseFrame

> error_code_value is the actual, numerical code (the parsed VLIE-encoded
> integer). This is useful because some error types are spread out over a range of
> codes (e.g., QUIC's crypto_error).

Reviewer: I find this a bit ambigous to read. How about something like

```suggestion
The error_code_value is the numerical value without VLIE encoding.
This is useful because some error types are spread out over a range of
codes (e.g., QUIC's crypto_error).
```

### UnknownFrame

frame_type_value is the actual, numerical code (the parsed VLIE-encoded

> integer).

Reviewer: Again, reads ambiguous to me.

> ## Since draft-ietf-qlog-quic-events-02:

Reviewer: ```suggestion
## Since draft-ietf-qlog-quic-events-03:
```
